File: 字符串/14.最长公共前缀.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) return "";
    strs.sort();
    //! 错误: strs.sort((a, b) => a - b);  字符串相减 NaN
    //! 正确: strs.sort(); 字典排序
    //! 正确: strs.sort((a, b) => a.localeCompare(b)); // 字典排序
    let s = strs[0];
    let l = strs[strs.length - 1];
    let i = 0;
    while (i < l.length) {
        if (s[i] !== l[i]) {
            break;
        }
        i++;
    }
    return s.substring(0, i);
};

File: 字符串/14.最长公共前缀.py
# 先排序    排序会将相关性最大最相似的从前往后排， 只用找第一个和最后一个的公共前缀
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        s, l, i = strs[0], strs[-1], 0
        while i < min(len(strs[-1]), len(strs[0])):
            if s[i] != l[i]:
                break
            i +=1
        return s[:i]